packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/funman/FunmanController.java

```diff
@@ -1,25 +1,97 @@
 package software.uncharted.terarium.hmiserver.controller.funman;
 
 import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import feign.FeignException;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
 import io.swagger.v3.oas.annotations.responses.ApiResponse;
 import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import jakarta.annotation.PostConstruct;
+import java.util.UUID;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.http.ResponseEntity;
 import org.springframework.security.access.annotation.Secured;
 import org.springframework.web.bind.annotation.*;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.server.ResponseStatusException;
 import software.uncharted.terarium.hmiserver.models.funman.FunmanPostQueriesRequest;
+import software.uncharted.terarium.hmiserver.models.task.TaskRequest;
+import software.uncharted.terarium.hmiserver.models.task.TaskRequest.TaskType;
+import software.uncharted.terarium.hmiserver.models.task.TaskResponse;
 import software.uncharted.terarium.hmiserver.proxies.funman.FunmanProxy;
 import software.uncharted.terarium.hmiserver.security.Roles;
+import software.uncharted.terarium.hmiserver.service.CurrentUserService;
+import software.uncharted.terarium.hmiserver.service.tasks.TaskService;
+import software.uncharted.terarium.hmiserver.service.tasks.TaskService.TaskMode;
+import software.uncharted.terarium.hmiserver.service.tasks.ValidateModelConfigHandler;
 
 @RestController
 @RequestMapping("/funman/queries")
 @RequiredArgsConstructor
 @Slf4j
 public class FunmanController {
 
+	private final ObjectMapper objectMapper;
+	private final TaskService taskService;
+	private final CurrentUserService currentUserService;
+
+	private final ValidateModelConfigHandler validateModelConfigHandler;
+
+	@PostConstruct
+	void init() {
+		taskService.addResponseHandler(validateModelConfigHandler);
+	}
+
+	@PostMapping("/test")
+	@Secured(Roles.USER)
+	@Operation(summary = "Dispatch a model configuration validation task")
+	@ApiResponses(
+			value = {
+				@ApiResponse(
+						responseCode = "200",
+						description = "Dispatched successfully",
+						content =
+								@Content(
+										mediaType = "application/json",
+										schema =
+												@io.swagger.v3.oas.annotations.media.Schema(
+														implementation = TaskResponse.class))),
+				@ApiResponse(
+						responseCode = "400",
+						description = "The provided document text is too long",
```

Comment:
What `document text` does FUNMAN needs? Or is this an artifact from GoLLM controller?

Reply:
I copy and pasted the swagger annotations. Will update.